packages/backend/src/telemetry/service.ts

```diff
@@ -32,25 +32,29 @@ export interface TelemetryService {
   ): void
 }
 
-interface TelemetryServiceDependencies extends BaseService {
+export interface TelemetryServiceDependencies extends BaseService {
   instanceName: string
   collectorUrls: string[]
   exportIntervalMillis?: number
   aseRatesService: RatesService
   internalRatesService: RatesService
   baseAssetCode: string
   baseScale: number
+  enableTelemetry: boolean
 }
 
 const METER_NAME = 'Rafiki'
 
 export function createTelemetryService(
   deps: TelemetryServiceDependencies
 ): TelemetryService {
+  if (!deps.enableTelemetry) {
+    return new NoopTelemetryServiceImpl(deps)
+  }
   return new TelemetryServiceImpl(deps)
 }
```

Comment:
Feel like I need to elaborate on this approach because it might deviate from the expectations in the issue a bit which said:

> In order to still respect the ENABLE_TELEMETRY and ENABLE_TELEMETRY_TRACES flags, all we can do is just to not register metric/trace providers in the telemetry initialization.

What we're doing here is adding a new implementation that just doesnt do anything and using that if telemetry is disabled. 

I can revise this if we don't like it but I think its working well.

I didn't do it as described for a couple reasons: 1) I don't think its possible - we arent registering metric/trace providers on service init. Traces are already registered outside the service so they seem to be a non-factor here. 2) I think even if we did we'd find ourselves in a scenario where we're calling methods like `incrementCounterWithTransactionAmount`, doing all that work (including network call) and then simply do nothing.


Alternatives considerd:
1. in the spirit of the original suggestion, we could simply gate the counter create in `getOrCreateCounter` (and histogram) behind if telemetry is enabled and returned undefined if disabled like:

```ts
  private getOrCreateCounter(name: string, options?: MetricOptions): Counter {
    let counter = this.counters.get(name)
    if (!counter) {
      counter = metrics.getMeter(METER_NAME).createCounter(name, options)
      this.counters.set(name, counter)
    }
    return counter
  }
  ```
  
  Then we just use optional chaining to do nothing if undefined:
  
  ```ts
    const counter = this.getOrCreateCounter(name)
    counter.?add(...)
  ```
  
But again, this has the problem of doing all the work in `incrementCounterWithTransactionAmount` for example only to not actually do anything at the end.

2. Just add an early return to `recordHistogram`/`incrementCounter` if telemetry is disabled. But would the person adding the next similar method remember to do the same?

Reply:
>  1) I don't think its possible - we arent registering metric/trace providers on service init

I see what you mean, what happens to the existing metrics/counters/histogramts when we keep the original/existing telemetry service, but with `enableTelemetry` disabled?

Reply:
Yes, I'm with you, having a No-op class is clearer, and we don't end up having any side effects (even if they don't really do anything).